Tests/PromiseProgressTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
```

Comment:
Most of these tests are good but we need test coverage for the generic Promise<T> as well as the non-generic promise.